src/odin/maneuversbuilder.cc

```diff
@@ -2421,8 +2519,14 @@ void ManeuversBuilder::ProcessRoundaboutNames(std::list<Maneuver>& maneuvers) {
       // Process roundabout exit names and signs
       if (next_man->type() == DirectionsLeg_Maneuver_Type_kRoundaboutExit) {
         if (next_man->HasBeginStreetNames()) {
-          curr_man->set_roundabout_exit_begin_street_names(next_man->begin_street_names().clone());
-          curr_man->set_roundabout_exit_street_names(next_man->street_names().clone());
+          if (next_man->contains_obvious_maneuver()) {
```

Comment:
Any concerns that this will cause issues when we merge it with #2437 ?

Reply:
No issues

Reply:
@danpaz I tested - this looks good